stl/inc/span

```diff
@@ -0,0 +1,665 @@
+};
+
+// CLASS TEMPLATE _Span_extent_type
+template <size_t _Extent>
+struct _Span_extent_type {
+    constexpr _Span_extent_type() noexcept = default;
+
+    constexpr explicit _Span_extent_type([[maybe_unused]] const size_t _Size) noexcept {
+#if _CONTAINER_DEBUG_LEVEL > 0
+        _STL_VERIFY(_Size == _Extent, "Mismatch between static extent and size of initializing data");
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+    }
+
+    template <size_t _OtherExtent>
+    constexpr explicit _Span_extent_type(const _Span_extent_type<_OtherExtent> _Ext) noexcept {
+        static_assert(_OtherExtent == _Extent || _OtherExtent == dynamic_extent,
+            "Mismatch between static extent and size of initializing data.");
+#if _CONTAINER_DEBUG_LEVEL > 0
+        if constexpr (_OtherExtent == dynamic_extent) {
+            _STL_VERIFY(_Ext._Size() == _Extent, "Mismatch between static extent and size of initializing data");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+    }
+
+    _NODISCARD constexpr size_t size() const noexcept {
+        return _Extent;
+    }
+};
+
+template <>
+struct _Span_extent_type<dynamic_extent> {
+    constexpr _Span_extent_type() noexcept = default;
+
+    constexpr explicit _Span_extent_type(const size_t _Size) noexcept : _Mysize(_Size) {}
+
+    template <size_t _OtherExtent>
+    constexpr explicit _Span_extent_type(const _Span_extent_type<_OtherExtent> _Ext) noexcept : _Mysize(_Ext._Size()) {}
+
+    _NODISCARD constexpr size_t size() const noexcept {
+        return _Mysize;
+    }
+
+private:
+    size_t _Mysize{0};
+};
+
+// VARIABLE TEMPLATE _Is_span_v
+template <class _Ty, size_t _Extent>
+class span;
+
+template <class>
+inline constexpr bool _Is_span_v = false;
+
+template <class _Ty, size_t _Extent>
+inline constexpr bool _Is_span_v<span<_Ty, _Extent>> = true;
+
+// VARIABLE TEMPLATE _Is_std_array_v
+template <class>
+inline constexpr bool _Is_std_array_v = false;
+
+template <class _Ty, size_t _Extent>
+inline constexpr bool _Is_std_array_v<array<_Ty, _Extent>> = true;
+
+// VARIABLE TEMPLATE _Is_span_convertible_range_v
+template <class _Cont, class _Ty>
+inline constexpr bool _Is_span_convertible_range_v =
+    is_convertible_v<remove_pointer_t<decltype(_STD data(_STD declval<_Cont&>()))> (*)[], _Ty (*)[]>;
+
+// VARIABLE TEMPLATE _Has_container_interface_v
+template <class, class = void>
+inline constexpr bool _Has_container_interface_v = false;
+
+template <class _Cont>
+inline constexpr bool _Has_container_interface_v<_Cont,
+    void_t<decltype(_STD data(_STD declval<_Cont&>())), decltype(_STD size(_STD declval<_Cont&>()))>> = true;
+
+// VARIABLE TEMPLATE _Is_span_compatible_container_v
+// clang-format off
+template <class _Cont, class _Ty>
+inline constexpr bool _Is_span_compatible_container_v =
+       !_Is_span_v<remove_const_t<_Cont>>
+    && !_Is_std_array_v<remove_const_t<_Cont>>
+    && !is_array_v<_Cont>
+    && _Is_span_convertible_range_v<_Cont, _Ty>
+    && _Has_container_interface_v<_Cont>;

```

Comment:
Because you're using `&&` instead of `conjunction_v`, there's no short-circuiting. This is a problem for the https://eel.is/c++draft/span.cons#14.5 constraint checked by `_Has_container_interface_v`. If it's false, then we need to avoid checking the https://eel.is/c++draft/span.cons#14.6 constraint with `_Is_span_convertible_range_v` (because forming the `data()` call could be ill-formed).

Reply:
Honestly I am havin trouble getting conjunction to work here gettings spuripus template errors

Reply:
Done. Although I am unsure, whether itis really clean in the way I did it only defining the `*_v` variable templates.

Also I would argue that `is_span` should be a user visible type trait

Reply:
I switched to struct templates so that `conjunction` works better